custom-completions/dotnet/README.md

```diff
@@ -9,6 +10,45 @@ For more information, see
 - [CLI documentation](https://learn.microsoft.com/en-us/dotnet/core/tools/),
 - [General dotnet documentation](https://learn.microsoft.com/en-us/dotnet/fundamentals/).
 
-This plugin uses built-in `dotnet complete` command, which unfortunately does not provide comments for the completions.
-On the other hand, it is officially supported, and completions are always in sync with the installed .NET SDK.
-For hand-crafted completions, see ones generated from Fish: (../auto-generate/completions/dotnet.nu).
+## SDK dependency and Limitations
+
+This plugin uses built-in `dotnet complete` command. Unfortunately, it does not cover all capabilities of Nushell, like completion comments.
+(See [capabilities](https://learn.microsoft.com/en-us/dotnet/core/tools/enable-tab-autocomplete#completion-capabilities).)
+
+Despite being the official, native, integrated completion mechanism, `dotnet complete` may not cover all commands.
+For example, version 10.0.100 does not cover the new `dotnet completions` commands.
+
+For hand-crafted completions, see other completions like the ones generated from Fish located in (../auto-generate/completions/dotnet.nu).
+
+## .NET SDK 10 and `dotnet completions`
+
+From the .NET SDK version 10 onwards, the `dotnet` CLI offers `dotnet completions script nushell` that generaltes a Nushell external completer configuration.
+Unfortunately, it is not directly pluggable, but more of a mix of generated forwarding completer and manual configuration integration guide:
+
+```nushell
+# Add the following content to your config.nu file:
+
+let external_completer = { |spans|
+    {
+        dotnet: { ||
+            dotnet complete (
+                $spans | skip 1 | str join " "
+            ) | lines
+        }
+    } | get $spans.0 | each { || do $in }
+}
+
+# And then in the config record, find the completions section and add the
+# external_completer that was defined earlier to external:
+
+let-env config = {
```

Comment:
`let-env` isn't a thing anymore. I noticed this bug in the microsoft documentation too.

Reply:
I'd like to know that it works before reporting upstream. I'm not in a place to test this right now but `let-env` definitely doesn't exist. It should be something like `$env.config` with merger or `++` or something to combine the config setting without totally overwriting the entire config like the current example does.